tests/config/test_search.py

```diff
@@ -0,0 +1,128 @@
+    expected.extend(base_search_paths())
+    assert search.config_search_paths(['~/my_configs']) == expected
+
+
+def test_default_returned_slot_1():
+    """Test config returned from ~/.atkinson"""
+    with patch('atkinson.config.search._check_available') as aval_mock:
+        aval_mock.return_value = True
+        assert search.get_config_file() == search.expand_user('~/.atkinson/config.yml')
+
+
+def test_default_returned_slot_2():
+    """Test config returned from /etc/atkinson"""
+    with patch('atkinson.config.search._check_available') as aval_mock:
+        aval_mock.side_effect = [False, True]
+        assert search.get_config_file() == '/etc/atkinson/config.yml'
+
+
+def test_default_returned_slot_3():
+    """Test config returned from ./config"""
+    with patch('atkinson.config.search._check_available') as aval_mock:
+        aval_mock.side_effect = [False, False, True]
+        assert search.get_config_file() == os.path.realpath('./configs/config.yml')
+
+
+def test_default_not_found():
+    """Test config can't be found in the default paths"""
+    with patch('atkinson.config.search._check_available') as aval_mock:
+        aval_mock.side_effect = [False, False, False]
+        assert search.get_config_file() == ''
+
+
+def test_custom_returned_slot_1():
+    """Test config returned from ~/.atkinson"""
+    with patch('atkinson.config.search._check_available') as aval_mock:
+        aval_mock.side_effect = [True]
+        expected = search.expand_user('~/.atkinson/my_config.yml')
+        assert search.get_config_file(filename='my_config.yml') == expected
+
+
+def test_custom_returned_slot_2():
+    """Test config returned from /etc/atkinson"""
+    with patch('atkinson.config.search._check_available') as aval_mock:
+        aval_mock.side_effect = [False, True]
+        assert search.get_config_file(filename='my_config.yml') == '/etc/atkinson/my_config.yml'
+
+
+def test_custom_returned_slot_3():
+    """Test config returned from ./config"""
+    with patch('atkinson.config.search._check_available') as aval_mock:
+        aval_mock.side_effect = [False, False, True]
+        expected = os.path.realpath('./configs/my_config.yml')
+        assert search.get_config_file(filename='my_config.yml') == expected
+
+
+def test_default_override():
+    """Test default config from override"""
+    with patch('atkinson.config.search._check_available') as aval_mock:
+        aval_mock.side_effect = [True]
+        expected = search.expand_user('~/my_config_dir/config.yml')
+        assert search.get_config_file(overrides='~/my_config_dir') == expected
+
+
+def test_default_override_not_found():
+    """Test defult from override list"""
+    with patch('atkinson.config.search._check_available') as aval_mock:
+        aval_mock.side_effect = [False, True]
+        expected = search.expand_user('~/.atkinson/config.yml')
+        assert search.get_config_file(overrides=['~/my_config_dir']) == expected
```

Comment:
What happens if you pass in two valid overrides?

Reply:
The first directory that containers the matching file will be taken. Let me rework that test to make it a little more clear.